api/src/services/translation.service.ts

```diff
@@ -192,9 +194,34 @@ export class TranslationService {
     listing: Listing,
     language: LanguagesEnum,
   ) {
-    return this.prisma.generatedListingTranslations.findFirst({
-      where: { listingId: listing.id, language: language },
+    const res = await this.prisma.generatedListingTranslations.findFirst({
+      where: {
+        listingId: listing.id,
+        language: language,
+      },
     });
+
+    //determine when listing or associated preferences most recently changed
+    let mostRecentUpdate = listing.contentUpdatedAt;
+    listing.listingMultiselectQuestions?.forEach((multiselectObj) => {
+      const multiselectUpdatedAt =
+        multiselectObj.multiselectQuestions?.updatedAt;
+      if (mostRecentUpdate < multiselectUpdatedAt) {
+        mostRecentUpdate = multiselectUpdatedAt;
+      }
+    });
+    //refresh translations if application content changed since translation creation
+    if (res && res.createdAt < mostRecentUpdate) {
+      await this.prisma.generatedListingTranslations.delete({
+        where: {
+          id: res.id,
+        },
+      });
+      console.log('get new translations');
+      return undefined;
```

Comment:
Does something need to be called? How does the translation update occur?

Reply:
Good q! It's called on [line 158](https://github.com/bloom-housing/bloom/blob/869697adc070b8ccc3c21bf29b0f5b1fbe1dc690/api/src/services/translation.service.ts#L158) and if it is defined we use them but if it is undefined we fetch new translations from the google translation service